### PR TITLE
Condition set view provider

### PR DESCRIPTION
**Pull request by JoelMcKinnon**
Completed ConditionSetViewProvider.

Changed patterns to emulate DisplayProvider per Pegah's suggestion.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -43,7 +43,7 @@
             openmct.legacyRegistry.enable.bind(openmct.legacyRegistry)
         );
 
-        openmct.install(openmct.plugins.Espresso());
+        openmct.install(openmct.plugins.Snow());
```
Comment (deeptailor) [inline index.html]: lets keep openmct default to espresso

File: src/plugins/conditionSet/ConditionSetViewProvider.js [hunk 2]
```diff
@@ -20,42 +20,46 @@
...
-import ConditionSetComponent from './components/ConditionSet.vue';
+import ConditionSet from './components/ConditionSet.vue';
 import Vue from 'vue';
 
-export default function ConditionSet(openmct) {
+export default function ConditionSetViewProvider(openmct) {
     return {
-        key: 'conditionSet',
-        name: 'Condition Set',
-        cssClass: 'icon-page',
```
Comment (deeptailor) [inline src/plugins/conditionSet/ConditionSetViewProvider.js]: I would keep the name and icon properties, they are used when more than one view is available for a type. Where we will allow the user to select from a drop down menu.

File: src/plugins/conditionSet/ConditionSetViewProvider.js [hunk 3]
```diff
@@ -20,42 +20,46 @@
...
-                            domainObject
+                            objectPath
+                        },
+                        data() {
+                            return {
+                                domainObject
+                            };
                         },
-                        template: '<condition-set-component></condition-set-component>'
+                        template: '<condition-set ref="conditionSet" :domain-object="domainObject"></condition-set>'
```
[the remark this answers is not in the document]
Comment (deeptailor) [reply src/plugins/conditionSet/ConditionSetViewProvider.js]: why are we storing a ref (reference) to this element?

[the remark this answers is not in the document]
Comment (JoelMcKinnon) [reply src/plugins/conditionSet/ConditionSetViewProvider.js]: Not sure, but was following pattern from displayLayout/plugin.js line 60. Why are we doing it there?

File: src/plugins/conditionSet/ConditionSetViewProvider.js [hunk 4]
```diff
@@ -20,42 +20,50 @@
...
-                            ConditionSetComponent
+                            ConditionSet
                         },
                         provide: {
                             openmct,
-                            domainObject
+                            domainObject,
+                            objectPath
+                        },
+                        data() {
```
Comment (deeptailor) [inline src/plugins/conditionSet/ConditionSetViewProvider.js]: @JoelMcKinnon This duplicates the data property in the component, i would remove from here

File: src/plugins/condition/ConditionSetViewProvider.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+ *****************************************************************************/
+
+import ConditionSet from './components/ConditionSet.vue';
+import Vue from 'vue';
+
+export default class ConditionSetViewProvider {
+    constructor(openmct) {
+        this.openmct = openmct;
+        this.key = 'conditionSet.view';
+        this.cssClass = 'icon-summary-widget';
```
Comment (shefalijoshi) [inline src/plugins/condition/ConditionSetViewProvider.js]: Can we add a TODO here to change this css later?

File: src/plugins/condition/components/Condition.vue [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+              class="condition-name"
+        >Default
+        </span>
+        <span v-else
+              class="condition-name"
+        >[condition name]
+        </span>
+        <span v-if="isDefault"
+              class="condition-output"
+        >Output: false
```
Comment (shefalijoshi) [inline src/plugins/condition/components/Condition.vue]: Should this output be false? Wouldn't `default` always be true?

Comment (JoelMcKinnon) [reply src/plugins/condition/components/Condition.vue]: I was replicating the UX mockup. To be honest I'm not sure what output would mean in this context. Let's discuss.

File: src/plugins/conditionSet/ConditionSetViewProvider.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (shefalijoshi) [reply src/plugins/conditionSet/ConditionSetViewProvider.js]: It's to get the reference for the add, remote and order item methods. We don't need this for conditions (yet, anyway). We can add it back later if we think we need it.

**Pull request merged**